ext/opcache/Optimizer/block_pass.c

```diff
@@ -1749,14 +1802,18 @@ static void zend_merge_blocks(zend_op_array *op_array, zend_cfg *cfg)
 				/* re-link */
 				prev->flags |= (b->flags & ZEND_BB_EXIT);
 				prev->len = b->start + b->len - prev->start;
-				prev->successors[0] = b->successors[0];
-				prev->successors[1] = b->successors[1];
+				prev->successors_count = b->successors_count;
+				if (b->successors != b->successors_storage) {
+					prev->successors = b->successors;
+					b->successors = b->successors_storage;
```

Comment:
I don't think this assignment is strictly necessary, but it's there to avoid having two different blocks pointing to the same successor list. I don't think anything bad can happen in this case (as `b` becomes unreachable), but this seems safer.